app/jobs/data_exports/create_job.rb

```diff
@@ -168,5 +189,72 @@ def write_samples_workflow_execution_outputs_and_manifest(swe, zip)
 
       @manifest['children'] << sample_directory
     end
+
+    # Linelist export specific functions---------------------------------------------------------------------
+    def create_linelist_spreadsheet(data_export)
+      namespace_type = Namespace.find(data_export.export_parameters['namespace_id']).type
+      samples = if namespace_type == Group.sti_name
+                  Sample.includes(project: :namespace)
+                        .where(id: data_export.export_parameters['ids'])
+                else
+                  Sample.where(id: data_export.export_parameters['ids'])
+                end
+      if data_export.export_parameters['format'] == 'csv'
+        write_csv_export(data_export, samples, namespace_type)
+      else
+        write_xlsx_export(data_export, samples, namespace_type)
+      end
+    end
+
+    def write_csv_export(data_export, samples, namespace_type)
+      Tempfile.new(binmode: true).tap do |tempfile|
+        CSV.open(tempfile, 'wb') do |csv|
+          csv << write_spreadsheet_header(namespace_type,
+                                          data_export.export_parameters['metadata_fields'])
+          samples.each do |sample|
+            csv << write_spreadsheet_row(sample, data_export, namespace_type)
+          end
+        end
+      end
+    end
+
+    def write_xlsx_export(data_export, samples, namespace_type)
+      Tempfile.new(binmode: true).tap do |tempfile|
+        Axlsx::Package.new do |workbook|
+          workbook.workbook.add_worksheet(name: 'linelist') do |sheet|
+            sheet.add_row write_spreadsheet_header(namespace_type,
+                                                   data_export.export_parameters['metadata_fields'])
+            samples.each do |sample|
+              sheet.add_row write_spreadsheet_row(sample, data_export, namespace_type)
+            end
+          end
+          workbook.serialize(tempfile.path)
+        end
+      end
+    end
+
+    def write_spreadsheet_header(namespace_type, metadata_fields)
+      header = %w[id sample]
```

Comment:
**nitpick (blocking):** Can you renamed `id` to `SAMPLE ID` and `sample` to `SAMPLE NAME`. Can you also rename `project` to `PROJECT ID` and display the `puid` instead of the path. The metadata_fields should also be in all caps for the header as well.

Reply:
Fixed in 091f97c97c9701cd65023b48d7eac2be556af4e2